### PR TITLE
Chapter and page are connected to url, page url also changes when the page is visible

### PR DESCRIPTION
**Pull request by fairnightzz**

### CODE REVIEW

File: src/components/MangaControl.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ const MangaControl: React.FunctionComponent<Props> = (props: Props) => {
     updateReaderSettings({...readerSettings, colourTheme: event.currentTarget.value as ColourTheme });
   };
   const changeChapter = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    console.log(event.currentTarget.value);
     updateChapter(+event.currentTarget.value);
```
Comment (MrPicklePinosaur) [inline src/components/MangaControl.tsx:33]: just make sure event.currentTarget.value is actually a number

**Review by MrPicklePinosaur — APPROVED** [review]
looks good to LGTM

File: src/pages/Reader.tsx [hunk 2]
```diff
@@ -114,17 +120,24 @@ class Reader extends React.Component<any, State> { // fix typing up
...
+    this.props.history.replace({ pathname: `/read/${this.state.mangafile}/${this.props.match.params.chapter}/${newPage+1}`})
     this.setState(curState => ({
       ...curState,
       page: newPage,
-    }))
+    }), () => !scroll && this.myRef?.current?.scrollIntoView())
+    
+  }
+
+  checkVisible = (inView: boolean, ind: number) => {
```
Comment (MrPicklePinosaur) [inline src/pages/Reader.tsx:137]: isn't this function a bit trivial

Comment (fairnightzz) [reply src/pages/Reader.tsx:137]: inView can return true or false

File: src/components/MangaControl.tsx (hunk 1 above)
Comment (fairnightzz) [reply src/components/MangaControl.tsx:33]: ok

Comment (fairnightzz) [reply src/components/MangaControl.tsx:33]: gdfg

Comment (fairnightzz) [reply src/components/MangaControl.tsx:33]: The thing is, changeChapter is called only when a chapter is selected, and the value of all those inputs are based on index, which is a number, so the error will never happen

**Pull request merged**